cmd_init_wallet.go

```diff
@@ -0,0 +1,484 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcwallet/wallet"
+	"github.com/jessevdk/go-flags"
+	"github.com/lightninglabs/protobuf-hex-display/jsonpb" // nolint
+	"github.com/lightningnetwork/lnd/aezeed"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnrpc/walletrpc"
+	"github.com/lightningnetwork/lnd/signal"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	defaultNoFreelistSync = true
+
+	defaultDirPermissions os.FileMode = 0700
+
+	defaultBitcoinNetwork = "mainnet"
+
+	typeFile = "file"
+	typeRpc  = "rpc"
+)
+
+var (
+	defaultWalletDBTimeout = 2 * time.Second
+)
+
+type secretSourceFile struct {
+	Seed           string `long:"seed" description:"The full path to the file that contains the seed; if the file does not exist, lndinit will exit with code EXIT_CODE_INPUT_MISSING (129)"`
+	SeedPassphrase string `long:"seed-passphrase" description:"The full path to the file that contains the seed passphrase; if not set, no passphrase will be used; if set but the file does not exist, lndinit will exit with code EXIT_CODE_INPUT_MISSING (129)"`
+	WalletPassword string `long:"wallet-password" description:"The full path to the file that contains the wallet password; if the file does not exist, lndinit will exit with code EXIT_CODE_INPUT_MISSING (129)"`
+}
+
+type secretSourceK8s struct {
+	Namespace               string `long:"namespace" description:"The Kubernetes namespace the secret is located in"`
+	SecretName              string `long:"secret-name" description:"The name of the Kubernetes secret"`
+	SeedEntryName           string `long:"seed-entry-name" description:"The name of the entry within the secret that contains the seed"`
+	SeedPassphraseEntryName string `long:"seed-passphrase-entry-name" description:"The name of the entry within the secret that contains the seed passphrase"`
+	WalletPasswordEntryName string `long:"wallet-password-entry-name" description:"The name of the entry within the secret that contains the wallet password"`
```

Comment:
In the example scripts it wasn't immediately obvious to me what `secret-entry-name` meant. I think renaming these from `*EntryName` to `*Key` or `*KeyName` would make it more intuitive to understand they refer to "the name of the key within the secret data" (adopting terminology from the Kubernetes docs).